### ISSUE TITLE
GRKBarGraphView

### ISSUE THREAD

**Issue by tvieweg** [levigroker/GRKBarGraphView#1]
Hi Levi, 

First off, thanks for this really handy bar graph view! I'm using it for a couple graphs in a project I'm developing currently. I'm sure this is user error, but when I use the code below, my bar graph view **monthlyDonationTotal** does not animate. For reference, these are IBOutlets from my storyboard: 
```objective-c
#import <GRKBarGraphView/GRKBarGraphView.h>

@implementation AccountSummaryViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.pageTitle.text = @"This is the first page";
    
    self.monthlyTransactionTotal.tintColor = [UIColor whiteColor];
    self.monthlyTransactionTotal.barStyle = GRKBarStyleFromBottom;
    [self.monthlyTransactionTotal setAnimationDuration:0.5];

    self.monthlyDonationTotal.tintColor = [UIColor whiteColor];
    self.monthlyDonationTotal.barStyle = GRKBarStyleFromBottom;
    self.monthlyDonationTotal.animationDuration = 1.0; 
    
}

- (void) viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    self.monthlyDonationTotal.percent = 1.0;

}

```

From the documentation you gave it looked like all I needed to do was set an animation time to trigger animations. Any idea what I'm doing wrong here? 

Any help is appreciated, thanks very much!

**Comment by tvieweg** [comment]
Resolved the issue, changing percent values needed to be called asynchronously.

**Comment by levigroker** [comment]
Hi Trevor,

You're very welcome. I hope you find it useful.

I'm not sure what you were experiencing, but setting the percent asynchronously should not be needed.

Does the test project work as expected in your environment?

Cheers,

Levi

**Comment by tvieweg** [comment]
Thanks Levi,

Yes it does, I'm running xCode 6.4 with a target 8.4 build. I ran the
example app and it seemed to work fine.

It seems like the animation is getting blocked up as the view loads. I set
up a test button in the view that when clicked would update the percent and
this worked as expected (no async call needed.) However when setting the
values as the view loads (in viewDidAppear) the async call was required.

In any case, it's now working for me. Just a heads up on a little kink.
Thanks again for the great tool!

-Trevor

On Mon, Aug 24, 2015 at 9:36 AM, Levi Brown <notifications@github.com>
wrote:

> Hi Trevor,
>
> You're very welcome. I hope you find it useful.
>
> I'm not sure what you were experiencing, but setting the percent
> asynchronously should not be needed.
>
> Does the test project work as expected in your environment?
>
> Cheers,
>
> Levi
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/levigroker/GRKBarGraphView/issues/1#issuecomment-134292626>
> .
>



-- 
------------------
Trevor Vieweg
trevor.vieweg@gmail.com
